diff --git a/package.json b/package.json
index c74c688e4..38e05bc79 100644
--- a/package.json
+++ b/package.json
@@ -4,6 +4,7 @@
     "packages/*",
     "website",
     "website-1.x",
+    "wener/site",
     "packages/docusaurus-init/templates/*"
   ],
   "scripts": {
diff --git a/packages/docusaurus-plugin-content-blog/src/blogUtils.ts b/packages/docusaurus-plugin-content-blog/src/blogUtils.ts
index 240ce1cc3..7d665c246 100644
--- a/packages/docusaurus-plugin-content-blog/src/blogUtils.ts
+++ b/packages/docusaurus-plugin-content-blog/src/blogUtils.ts
@@ -105,6 +105,7 @@ export async function generateBlogPosts(
       const blogFileName = path.basename(relativeSource);
 
       const fileString = await fs.readFile(source, 'utf-8');
+      if (!fileString.startsWith('--')) return;
       const {frontMatter, excerpt} = parse(fileString);
 
       let date;
diff --git a/packages/docusaurus-plugin-content-docs/src/index.ts b/packages/docusaurus-plugin-content-docs/src/index.ts
index d5185d123..850b9af97 100644
--- a/packages/docusaurus-plugin-content-docs/src/index.ts
+++ b/packages/docusaurus-plugin-content-docs/src/index.ts
@@ -142,6 +142,7 @@ export default function pluginContentDocs(
               options,
               env,
             });
+            if(metadata && metadata.id)
             docsMetadataRaw[metadata.id] = metadata;
           }),
         ),
diff --git a/packages/docusaurus-plugin-content-docs/src/metadata.ts b/packages/docusaurus-plugin-content-docs/src/metadata.ts
index 873c3bca0..a0442aa5e 100644
--- a/packages/docusaurus-plugin-content-docs/src/metadata.ts
+++ b/packages/docusaurus-plugin-content-docs/src/metadata.ts
@@ -93,7 +93,9 @@ export default async function processMetadata({
     ? normalizeUrl([editUrl, posixPath(relativePath)])
     : undefined;
 
-  const {frontMatter = {}, excerpt} = parse(await fileStringPromise);
+  let fileContent = await fileStringPromise;
+  if(!fileContent.startsWith("--"))return {} as MetadataRaw;
+  const {frontMatter = {}, excerpt} = parse(fileContent);
   const {sidebar_label, custom_edit_url} = frontMatter;
 
   // Default base id is the file name.
