diff --git a/lib/docs.js b/lib/docs.js
index 407677cf81499599584045ba327083ba77d3bbec..722f20f98a14d9d4f8bccb930bee27d9f098073e 100644
--- a/lib/docs.js
+++ b/lib/docs.js
@@ -123,7 +123,7 @@ async function doProcessDocMetadata({ docFile, versionMetadata, context, options
         frontMatterSlug: frontMatter.slug,
         stripDirNumberPrefixes: parseNumberPrefixes,
         numberPrefixParser: options.numberPrefixParser,
-    });
+    }).replace(/\/([^\/]+)\/([^\/]+)\/\1-\2-/, '/$1/$2/').replace(/\/([^/]+)\/\1-/,'/$1/').replace(/\/(kubernetes)\/k8s-/,'/$1/');
     // Note: the title is used by default for page title, sidebar label,
     // pagination buttons... frontMatter.title should be used in priority over
     // contentTitle (because it can contain markdown/JSX syntax)
diff --git a/lib/index.js b/lib/index.js
index b74c7f92a28b240696103758c684e6ebdae88ed1..685bec6a384c5d3b6770cb7d65916941b2cfb791 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -83,7 +83,8 @@ async function pluginContentDocs(context, options) {
                         env: process.env.NODE_ENV,
                     });
                 }
-                return Promise.all(docFiles.map(processVersionDoc));
+                console.log(`Inject custom docs filter`);
+                return Promise.all(docFiles.filter(v=>v.content.startsWith('--')).map(processVersionDoc));
             }
             async function doLoadVersion(versionMetadata) {
                 const docsBase = await loadVersionDocsBase(versionMetadata);
